### PR TITLE
Update the home page of the User Manual (userguide.adoc) and fix all doc build errors

### PR DESCRIPTION
**Pull request by lkasso**
### Details
1. This PR fixes all the warnings when building the documentation.
2. This PR updates the home page of the user manual: https://docs.gradle.org/current/userguide/userguide.html

### Context
1. There are many warnings when building the docs in Gradle. This commit fixes all of them and creates a clean output:
Images
2. The [home page of the user manual](https://docs.gradle.org/current/userguide/userguide.html) is now a fusion of the what_is_gradle.adoc and userguide.adoc. The what_is_gradle.adoc is deleted and had been updated in https://github.com/gradle/gradle/pull/24831. 

### Reviewers
1. Nothing to view.
2. The new home page is hosted on builds.gradle.org --> [here](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/66741781:id/distributions/gradle-8.3-docs.zip!/gradle-8.3-20230606180940%2B0000/docs/userguide/userguide.html) and ready to merge.

### CODE REVIEW

**Review by tresat — CHANGES_REQUESTED** [review]
I left a comment about the redesign, I think it needs further consideration.

**Comment by akshayabd** [comment]
I'm still reviewing the main content changes, just wanted to +1 to what Tom said about having separate PRs for different types of changes - ie. fixing the warnings could be in its own PR.

File: subprojects/docs/src/docs/userguide/img/userguide-languages.png
Comment (akshayabd) [inline subprojects/docs/src/docs/userguide/img/userguide-languages.png:1]: This image is missing Kotlin and Groovy. I see you moved them to the DSL section but I think its still worth displaying them here.

Also while Android isn't a language I think we do want to mention that Gradle is the main build tool for the Android community and display the logo

File: subprojects/docs/src/docs/userguide/userguide.adoc [hunk 1]
```diff
@@ -14,55 +14,35 @@
...
-</div>
-++++
+image::userguide-languages.png[]
 
-== New projects with Gradle
+== Supported IDEs
+Gradle is supported on Android Studio, IntelliJ IDEA, Eclipse Buildship, and Netbeans.
```
Comment (akshayabd) [inline subprojects/docs/src/docs/userguide/userguide.adoc:30]: Hmmm I know the existing page uses the word "support" but I'm not sure it's the correct word to use here. Gradle has integrations with these editors to provide a better/more seem-less experience.

I also think its worth still worth mentioning the CLI. In fact not only mentioning it but saying that its robust/comprehensive/fully-featured or something like that. There are people who prefer using command-line and we want to let them know there is good support for their use-cases

File: subprojects/docs/src/docs/userguide/userguide.adoc [hunk 2]
```diff
@@ -14,55 +14,35 @@
...
-== Getting help
-
- * **Forum** — The fastest way to get help is through the link:https://discuss.gradle.org[Gradle Forum, title="Gradle help and discussion forums"]. Community members and core contributors answer your questions.
- * **Training** — Free, web-based Gradle training from Gradle developers happens every month. Head over to the link:https://gradle.org/training/[training page, title="Gradle training schedule"] to sign up.
- * **Enterprise Services** — Support and training can be purchased alongside a link:https://gradle.com[Gradle Enterprise] subscription.
+== Documentation and Support
+ * **Gradle User Manual** — This <<about_manual.adoc#about_manual,User Manual>> is the official documentation for the Gradle Build Tool.
+ * **Forum** — The fastest way to get help is through the link:https://discuss.gradle.org/[Gradle Forum].
+ * **Slack** — Community members and core contributors answer questions directly on our link:https://gradle-community.slack.com/[Slack Channel].
+ * **Training** — Head over to the link:https://gradle.org/courses/[courses page] to sign up for free Gradle training.
```
Comment (akshayabd) [inline subprojects/docs/src/docs/userguide/userguide.adoc:43]: I like the addition of Slack here

**Comment by lkasso** [comment]
@tresat and @akshayabd, thank you for the feedback.

The original intent of this PR was to actually remove this page entirely as part of a cleanup effort. Which I would still prefer. 
- Remove: https://docs.gradle.org/nightly/userguide/userguide.html because it is low value to our users.
- Make https://docs.gradle.org/nightly/userguide/what_is_gradle.html the home page instead.

I ended up updating the page instead because there are many references to it all over our site and the web. 

I would like delete `userguide.adoc` and rename `what_is_gradle` to `userguide.adoc`.

@tresat @akshayabd thoughts?

**Comment by tresat** [comment]
> **I would like delete `userguide.adoc` and rename `what_is_gradle` to `userguide.adoc`**. @tresat @akshayabd thoughts?

I like the idea of having just one doc instead of those 2 separate ones.  Starting from What is Gradle and adding some of the useful content from the User Guide page in new sections below it could work.  Currently, I think What is Gradle is too short and text-only for it to be the landing page, but I agree the content there is a much better introduction for a newcomer to the tool.

**Comment by lkasso** [comment]
> I like the idea of having just one doc instead of those 2 separate ones. Starting from What is Gradle and adding some of the useful content from the User Guide page in new sections below it could work. Currently, I think What is Gradle is too short and text-only for it to be the landing page, but I agree the content there is a much better introduction for a newcomer to the tool.
Ok, perfect, I will check this with @pioterj and update the PR accordingly.

File: subprojects/docs/src/docs/userguide/userguide.adoc [hunk 3]
```diff
@@ -14,55 +14,35 @@
...
 = Gradle User Manual
 
-TIP: Want to learn the tips and tricks top engineering teams use to keep builds fast and performant? https://gradle.org/training/#build-cache-deep-dive[Register here] for our Build Cache Training.
+Gradle Build Tool is a fast, dependable, and adaptable open-source build automation tool with an elegant and extensible declarative build language:
 
-Gradle is an open-source build automation tool focused on flexibility and performance. Gradle build scripts are written using a link:https://groovy-lang.org/[Groovy] or link:https://kotlinlang.org/[Kotlin] DSL. Read about link:{website}/features/[Gradle features] to learn what is possible with Gradle.
+1. Write your App in Java, Kotlin, C++, or any language of your choice.
+2. Use Gradle to package your App for deployment on any platform.
+3. Use Gradle's APIs, plugins, and integrations to automate your build.
+4. Use Gradle's advanced features like caching to scale and optimize your build.
```
[the remark this answers is not in the document]
Comment (lkasso) [reply subprojects/docs/src/docs/userguide/userguide.adoc]: This has been removed.

File: subprojects/docs/src/docs/userguide/userguide.adoc (hunk 1 above)
Comment (lkasso) [reply subprojects/docs/src/docs/userguide/userguide.adoc]: Updated in latest commit.

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle clean

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
> > I like the idea of having just one doc instead of those 2 separate ones. Starting from What is Gradle and adding some of the useful content from the User Guide page in new sections below it could work. Currently, I think What is Gradle is too short and text-only for it to be the landing page, but I agree the content there is a much better introduction for a newcomer to the tool.
> 
> Ok, perfect. I will check this with @pioterj and update the PR accordingly.

PR has been updated accordingly and ready to merge.

**Comment by lkasso** [comment]
@bot-gradle clean and test BD

**Comment by lkasso** [comment]
@bot-gradle test BD

File: subprojects/docs/src/docs/userguide/getting_started.adoc [hunk 4]
```diff
@@ -19,7 +19,7 @@ Everyone has to start somewhere and if you're new to Gradle, this is where to be
 
 == Before you start
 
-In order to use Gradle effectively, you need to know what it is and understand some of its fundamental concepts. So before you start using Gradle in earnest, we highly recommend you read <<what_is_gradle#what_is_gradle,What is Gradle?>>.
+In order to use Gradle effectively, you need to know what it is and understand some of its fundamental concepts. So before you start using Gradle in earnest, we highly recommend you read <<userguide#what_is_gradle,What is Gradle?>>.
```
Comment (tresat) [inline subprojects/docs/src/docs/userguide/getting_started.adoc:22]:
```suggestion
In order to use Gradle effectively, you need to know what it is and understand some of its fundamental concepts. So before you start using Gradle in earnest, we highly recommend you read <<userguide#what_is_gradle,What is Gradle?>>
```

**Review by tresat — APPROVED** [review]
Preview [the new page here](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/66687513:id/distributions/gradle-8.3-docs.zip!/gradle-8.3-20230606042215%2B0000/docs/userguide/userguide.html).

Overall, I like it and think it is an improvement.

A few general comments:
- Seems like a lot of empty space, especially compared to previous page.  Can the margins after the headers be shrunk?
- Putting a Gradlephant logo (perhaps the one wielding the hammer from the WM presentation) somewhere might be nice, especially if sized slighly larger than the IDE and framework logos, to focus attention on the fact this is a Gradle site.
- Should the options of Groovy vs. Kotlin DSLs be highlighted anywhere?  I think it's probably okay to leave them out.
- There's no call to action anywhere - I would expect this would be a link to the Getting Started page.

File: subprojects/docs/src/docs/userguide/getting_started.adoc (hunk 4 above)
Comment (lkasso) [reply subprojects/docs/src/docs/userguide/getting_started.adoc:22]: Hey @tresat, thank you. 

Notes:
- The Getting Started page and Quick Start section will be updated entirely in https://github.com/gradle/gradle/pull/24913.
- When the Getting Started / Quick Start section is updated, I will be able to add the links in the new home page (I can't do it now or I will get dead link errors in the build - I will do it in PR [24913](https://github.com/gradle/gradle/pull/24913) instead).

Comment (tresat) [reply subprojects/docs/src/docs/userguide/getting_started.adoc:22]: Sounds good!

**Comment by lkasso** [comment]
@bot-gradle cleanup

**Comment by lkasso** [comment]
@bot-gradle clean

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle test BD

**Review by akshayabd — APPROVED** [review]

**Comment by lkasso** [comment]
@bot-gradle test and merge

**Pull request merged**